documentation/getting-started.adoc

```diff
@@ -0,0 +1,245 @@
+
+To deploy Kafka Connect on Kubernetes, the corresponding ConfigMap has to be created. An example ConfigMap can be
+created using the following command:
+
+[source]
+kubectl apply -f resources/kubernetes/kafka-connect.yaml
+
+==== Deploying to OpenShift
+
+On OpenShift, Kafka Connect is provided in the form of a template. It can be deployed from the template either
+using the command line or using the OpenShift console. To create Kafka Connect cluster from the command line, the following
+command should be run:
+
+[source]
+oc new-app strimzi-connect
+
+==== Using Kafka Connect with additional plugins
+
+Strimzi Docker images for Kafka Connect contain, by default, only the `FileStreamSinkConnector` and
+`FileStreamSourceConnector` connectors which are part of the Apache Kafka project.
+
+To facilitate deployment with 3rd party connectors, Kafka Connect is configured to automatically load all
+plugins/connectors which are present in the `/opt/kafka/plugins` directory during startup. There are two ways how to add
+custom plugins into this directory:
+
+- Using a custom Docker image
+- Using the OpenShift build system with the Strimzi S2I image
+
+===== Create a new image based on `strimzi/kafka-connect`
+
+Strimzi provides its own Docker image for running Kafka Connect which can be found on Docker Hub as
+https://hub.docker.com/r/strimzi/kafka-connect/[`strimzi/kafka-connect`]. This image could be used as a base image for
+building a new custom image with additional plugins. The following steps describe the process for creating such a custom image:
+
+1. Create a new `Dockerfile` which uses `strimzi/kafka-connect` as the base image
+
+[source,Dockerfile]
+FROM strimzi/kafka-connect:latest
+USER root:root
+COPY ./my-plugin/ /opt/kafka/plugins/
+USER kafka:kafka
+
+2. Build the Docker image and upload it to the appropriate Docker repository
+3. To use the new Docker image in the Kafka Connect deployment,
+  - On OpenShift, the template parameters `IMAGE_REPO_NAME`, `IMAGE_NAME` and `IMAGE_TAG` can be changed to point to the
+  new image when the Kafka Connect cluster is being deployed
+  - On Kubernetes, the Kafka Connect ConfigMap has to be modified to use the new image
+
+===== Using OpenShift Build and S2I image
+
+OpenShift supports https://docs.openshift.org/3.6/dev_guide/builds/index.html[Builds] which can be used together with
+https://docs.openshift.org/3.6/creating_images/s2i.html#creating-images-s2i[Source-to-Image (S2I)] framework to create
+new Docker images. OpenShift Build takes a builder image with S2I support together with source code and/or binaries
+provided by the user and uses them to build a new Docker image. The newly created Docker Image will be stored in
+OpenShift's local Docker repository and can then be used in deployments. The Strimzi project provides a Kafka Connect builder
+image https://hub.docker.com/r/strimzi/kafka-connect-s2i/[`strimzi/kafka-connect-s2i`] with S2I support. It takes user-provided
```

Comment:
...with such S2I support.

That makes the link to the previous sentence clearer